### PR TITLE
add external port object type

### PR DESCRIPTION
**Pull request by godber**
This changes `external_ports` from a simple array of numbers to an array that can contain numbers or objects, looks like:

```json
    "external_ports": [
        9090,
        {"name": "metrics", "port": 3333}
    ]
```

### CODE REVIEW

File: packages/job-components/src/job-schemas.ts [hunk 1]
```diff
@@ -254,17 +255,44 @@ export function jobSchema(context: Context): convict.Schema<any> {
...
+                            Object.entries(portValue).forEach(([key, val]) => {
+                                if (key == null || key === '') {
+                                    throw new Error('key must be not empty');
+                                }
+
+                                if (val == null || val === '') {
+                                    throw new Error(`value for key "${key}" must be not empty`);
+                                }
+                                // eslint-disable-next-line no-prototype-builtins
+                                if (portValue.hasOwnProperty('name') && portValue.hasOwnProperty('port')) {
```
Comment (peterdemartini) [inline packages/job-components/src/job-schemas.ts:275]: That eslint rule shouldn't be ignored, I would recommend using `hasOwn` from `@terascope/utils`.

```suggestion
                                if (hasOwn(portValue, 'name') && hasOwn(portValue, 'port')) {
```

**Review by peterdemartini — CHANGES_REQUESTED** [review]
One small change, then we can bump this

File: packages/job-components/src/job-schemas.ts (hunk 1 above)
Comment (godber) [reply packages/job-components/src/job-schemas.ts]: I've addressed this and bumped the patch version.

**Review by peterdemartini — APPROVED** [review]
LGTM

**Comment by godber** [comment]
This PR is also relevant to https://github.com/terascope/teraslice/issues/2608

**Pull request merged**